src/System.CommandLine/Binding/ServiceProviderValueSource.cs

```diff
@@ -5,12 +5,12 @@ namespace System.CommandLine.Binding
 {
     internal class ServiceProviderValueSource : IValueSource
     {
-        public bool TryGetValue(
-            IValueDescriptor valueDescriptor,
-            BindingContext? bindingContext,
-            out object? boundValue)
+        public bool TryGetValue(IValueDescriptor valueDescriptor,
+                                BindingContext? bindingContext,
+                                out object? boundValue)
         {
             boundValue = bindingContext?.ServiceProvider.GetService(valueDescriptor.ValueType);
+            // ?? Why return true if the service isn't found?
```

Comment:
This is because `null` is an acceptable return value for `ServiceProvider.GetService`.

Reply:
OK.

This led to me reviewing how we are using the return value in other ValueSources, and I think we should separately look at the name "ParseResultMatchingValueSource". It seems a name closer to MatchByName might be helpful

Reply:
Yeah, that seems helpful in the name, as is the fact that it matches from within the parse results, not e.g. the service provider.